src/SourceCode.Clay.OpenApi/ApiKeySecurityScheme.cs

```diff
@@ -0,0 +1,84 @@
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Gets the location of the API key.
+        /// </summary>
+        public ParameterLocation Location { get; }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="ApiKeySecurityScheme"/> class.
+        /// </summary>
+        /// <param name="description">The short description for security scheme.</param>
+        /// <param name="name">The name of the header, query or cookie parameter to be used.</param>
+        /// <param name="location">The location of the API key.</param>
+        public ApiKeySecurityScheme(
+            string description = default,
+            string name = default,
+            ParameterLocation location = default)
+            : base(description)
+        {
+            Name = name;
+            Location = location;
+        }
+
+        #region Equatable
+        /// <summary>Determines whether the specified object is equal to the current object.</summary>
+        /// <param name="obj">The object to compare with the current object.</param>
+        /// <returns>true if the specified object  is equal to the current object; otherwise, false.</returns>
+        public override bool Equals(object obj) => Equals(obj as ApiKeySecurityScheme);
+
+        /// <summary>Indicates whether the current object is equal to another object of the same type.</summary>
+        /// <param name="other">An object to compare with this object.</param>
+        /// <returns>true if the current object is equal to the <paramref name="other">other</paramref> parameter; otherwise, false.</returns>
+        public bool Equals(ApiKeySecurityScheme other)
+        {
+            if (!base.Equals(other)) return false;
+            if (!StringComparer.Ordinal.Equals(Name, other.Name)) return false;
+            if (Location != other.Location) return false;
+
+            return true;
+        }
+
+        /// <summary>Serves as the default hash function.</summary>
+        /// <returns>A hash code for the current object.</returns>
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hc = 17L;
+
+                hc = hc * 21 + base.GetHashCode();
+                if (Name != null) hc = hc * 21 + StringComparer.Ordinal.GetHashCode(Name);
+                hc = hc * 21 + Location.GetHashCode();
+
+                return ((int)(hc >> 32)) ^ (int)hc;
+            }
+        }
+
+        public static bool operator ==(ApiKeySecurityScheme scheme1, ApiKeySecurityScheme scheme2)
+        {
+            if (ReferenceEquals(scheme1, null) && ReferenceEquals(scheme2, null)) return true;
+            if (ReferenceEquals(scheme1, null) || ReferenceEquals(scheme2, null)) return false;
+            return scheme1.Equals((object)scheme2);
+        }
+
+        public static bool operator !=(ApiKeySecurityScheme scheme1, ApiKeySecurityScheme scheme2)
```

Comment:
Our existing conventions use (x, y)

Reply:
The equality refactoring has changed to use `foo1` and `foo2`. We should likely use that as it makes our lives easier in the long run.